From 6d4b755295896f0075ceb1787a10866a406f0d9a Mon Sep 17 00:00:00 2001
From: Piotr Marcinkowski <piotr.marcinkowski@redembedded.com>
Date: Tue, 28 Feb 2023 11:51:16 +0000
Subject: [PATCH] OMWAPPI-910 Increase subsamples limit for 4k content

The existing size of the array that holds sub-sample info structures
was too low. The number of sub-samples observed while playing 4k content
was bigger than that with maximum value of 160 - that value was doubled
and finally the array size grew from 100 to 320.

This solution isn't perfect - specification allows the number of
sub-samples to be bigger (16bit value)
---
 interfaces/IContentDecryption.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/interfaces/IContentDecryption.h b/interfaces/IContentDecryption.h
index f1b4481..13ca11a 100644
--- a/interfaces/IContentDecryption.h
+++ b/interfaces/IContentDecryption.h
@@ -69,7 +69,7 @@ namespace Exchange {
             uint32_t PatternClearBlocks;
             uint8_t  IV[24];
             uint8_t  KeyId[17];
-            CDMi::SubSampleInfo SubSamples[100];
+            CDMi::SubSampleInfo SubSamples[320];
             uint16_t StreamHeight;
             uint16_t StreamWidth;
             uint8_t  StreamType;
-- 
2.39.0

